include/picongpu/particles/shapes/TSC.hpp

```diff
@@ -128,6 +182,17 @@ namespace picongpu
 
                         return result;
                     }
+
+                    /** Creates an array with assignment values.
+                     *
+                     * @param pos particle position relative to the assignment cell range [-0.5;0.5)
+                     * @param isOutOfRange input will be ignored because the particle shape is always on support.
+                     * @return Array with precomputed assignment values.
+                     */
+                    HDINLINE auto shapeArray(float_X const x, [[maybe_unused]] bool const isOutOfRange) const
```

Comment:
This is logically consistent only with `isOutOfRange = false`. So we should at least add this to the comment, and potneetially also as an assert

Reply:
added a comment